storage/innobase/include/srv0tmp.h

```diff
@@ -307,6 +312,10 @@ class Tablespace_pool {
   Pool *m_active;
   /** Mutex to protect concurrent operations on the pool */
   ib_mutex_t m_mutex;
+
+  /** Next available space_id for tablespace. These are
+  hardcoded space_ids at higher range */
+  static space_id_t m_last_used_space_id;
```

Comment:
Shouldn't we use some kind of protection for this `m_last_used_space_id` (atomic var or mutex). Or even for the whole `Tablespace_pool::expand()` function body.
As far as I understood `Tablespace_pool::expand()` is called from `Tablespace_pool::get()` which in turn can be called concurrently?

Reply:
There is protection yura:   ut_ad(!m_pool_initialized || mutex_own(&m_mutex)); -> at beginning of function

Reply:
Oops, sorry. My bad :(
There is indeed

Reply:
np yura :)